server/server.go

```diff
@@ -1002,42 +1002,73 @@ func (s *Server) getClientConnectURLs() []string {
 	sPort := strconv.Itoa(opts.Port)
 	urls := make([]string, 0, 1)
 
-	ipAddr, err := net.ResolveIPAddr("ip", opts.Host)
-	// If the host is "any" (0.0.0.0 or ::), get specific IPs from available
-	// interfaces.
-	if err == nil && ipAddr.IP.IsUnspecified() {
-		var ip net.IP
-		ifaces, _ := net.Interfaces()
-		for _, i := range ifaces {
-			addrs, _ := i.Addrs()
-			for _, addr := range addrs {
-				switch v := addr.(type) {
-				case *net.IPNet:
-					ip = v.IP
-				case *net.IPAddr:
-					ip = v.IP
-				}
-				// Skip non global unicast addresses
-				if !ip.IsGlobalUnicast() || ip.IsUnspecified() {
-					ip = nil
-					continue
+	// short circuit if cluster-advertise is set
+	if opts.Cluster.ClientAdvertiseStr != "" {
+		hosts := strings.Split(opts.Cluster.ClientAdvertiseStr, ",")
+
+		for n, i := range hosts {
+			host, port, err := net.SplitHostPort(i)
+			switch err.(type) {
+			case *net.AddrError:
+				// try appending the current port
+				host, port, err = net.SplitHostPort(i + ":" + sPort)
+			}
+
+			if err != nil {
+				s.Fatalf("Client Advertise Address error: %v, on entry: %s", err, i)
+			}
+
+			// set the info host to the first address in a list
+			if n == 0 {
```

Comment:
Don't do this. As of now, routing uses the server's info's host and port to construct the URL for the accepted route. You may break things. I would say this is what cluster advertise will be used for.

Reply:
@kozlovic how will this work in the case of Clients that are proxied? What could this break? Today, if you're listening on all hosts, it returns `0.0.0.0`. I'll look at the popular clients to see if any currently do anything with that address.

Reply:
Clients don't use server's INFO's host/port at all.